hab/hab.go

```diff
@@ -0,0 +1,96 @@
+package hab
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// PackagesInfo is response from depotj
+type PackagesInfo struct {
+	RangeStart  int           `json:"range_start"`
+	RangeEnd    int           `json:"range_end"`
+	TotalCount  int           `json:"total_count"`
+	PackageList []PackageInfo `json:"package_list"`
+}
+
+// PackageInfo is package info in pkgs response
+type PackageInfo struct {
+	Origin  string `json:"origin"`
+	Name    string `json:"name"`
+	Version string `json:"version"`
+	Release string `json:"release"`
+}
+
+// Depot for hab
+type Depot interface {
+	PackageVersionsFromName(pkgName string) ([]string, error)
+}
+
+type depot struct {
+	baseURL string
+	client  *http.Client
+}
+
+// New returns a new depot object
+func New(baseURL string) Depot {
+	return &depot{baseURL, &http.Client{Timeout: 10 * time.Second}}
+}
+
+// packagesInfo fetch packages info from depot
+func (depo *depot) packagesInfo(pkgName string, from int) (PackagesInfo, error) {
+	pkgURL := fmt.Sprintf("%s/pkgs/%s?range=%d", depo.baseURL, pkgName, from)
+	res, err := depo.client.Get(pkgURL)
+
+	if err != nil {
+		return PackagesInfo{}, err
+	}
+
+	defer res.Body.Close()
+
+	if res.StatusCode == 404 {
+		return PackagesInfo{}, errors.New("Package not found")
+	} else if res.StatusCode != 200 {
+		return PackagesInfo{}, fmt.Errorf("Unexpected status code: %d", res.StatusCode)
+	}
+
+	var pkgsInfo PackagesInfo
+	err = json.NewDecoder(res.Body).Decode(&pkgsInfo)
+
+	if err != nil {
+		return PackagesInfo{}, err
+	}
+
+	return pkgsInfo, nil
+}
+
+// PackageVersionsFromName fetch all versions from depot
+func (depo *depot) PackageVersionsFromName(pkgName string) ([]string, error) {
+	var packages []PackageInfo
+
+	offset := 0
+	for {
+		pkgsInfo, err := depo.packagesInfo(pkgName, offset)
+
+		if err != nil {
+			return nil, err
+		}
+
+		packages = append(packages, pkgsInfo.PackageList...)
+
+		offset = pkgsInfo.RangeEnd + 1
+
+		if offset >= pkgsInfo.TotalCount {
+			break
+		}
+	}
+
+	var versions []string
+	for _, pkg := range packages {
+		versions = append(versions, pkg.Version)
```

Comment:
Does this handle duplication?
For example, this returns 18 packages but actually only 4 versions https://willem.habitat.sh/v1/depot/pkgs/core/redis?range=0

If it's easy to add without duplication, it would be good. Since the list to sort will be shorter. If it's too complicated, then let's not worry about it.